src/pages/DelegatoryValidator/index.tsx

```diff
@@ -24,32 +26,71 @@ export default function ValidatorPage() {
     "0x1::stake::StakePool",
   );
 
+  const [isDetailCardSkeletonLoading, setIsDetailedCardSkeletonLoading] =
+    useState<boolean>(true);
+  const [
+    isMyDepositsSectionSkeletonLoading,
+    setIsMyDepositsSectionSkeletonLoading,
+  ] = useState<boolean>(true);
+  const [isStakingBarSkeletonLoading, setIsStakingBarSkeletonLoading] =
+    useState<boolean>(true);
+  const [isSkeletonLoading, setIsSkeletonLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (
+      !isDetailCardSkeletonLoading &&
+      !isMyDepositsSectionSkeletonLoading &&
+      !isStakingBarSkeletonLoading
+    ) {
+      setIsSkeletonLoading(false);
+    }
+  }, [
+    isDetailCardSkeletonLoading,
+    isMyDepositsSectionSkeletonLoading,
+    isStakingBarSkeletonLoading,
+  ]);
+
```

Comment:
i would argue the skeleton logic should be taken out and written in a hook or something. right now the code is not very clean. low-pri though. can we add a BE todo here?